include/seqan3/range/view/interleave.hpp

```diff
@@ -0,0 +1,352 @@
+             std::Same<reference_t<urng_t>, reference_t<inserted_rng_t>>
+    //!\endcond
+class view_interleave : public ranges::view_interface<view_interleave<urng_t, inserted_rng_t>>
+{
+private:
+    //!\brief The underlying range.
+    urng_t urange;
+    //!\brief The step size for the insertion.
+    size_t step_size{};
+    //!\brief The range to be inserted into urange.
+    inserted_rng_t inserted_range;
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief The reference_type.
+    using reference         = reference_t<urng_t>;
+    //!\brief The const_reference type is equal to the reference type.
+    using const_reference   = detail::transformation_trait_or_t<seqan3::reference<urng_t const>, void>;
+    //!\brief The value_type (which equals the reference_type with any references removed).
+    using value_type        = value_type_t<urng_t>;
+    //!\brief If the underliying range is Sized, this resolves to range_type::size_type, otherwise void.
+    using size_type         = size_type_t<urng_t>;
+    //!\brief A signed integer type, usually std::ptrdiff_t.
+    using difference_type   = difference_type_t<urng_t>;
+    //!\brief The iterator type of this view (a random access iterator).
+    using iterator          = detail::random_access_iterator<view_interleave>;
+    //!\brief The const_iterator type is equal to the iterator type.
+    using const_iterator    = detail::random_access_iterator<view_interleave const>;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    view_interleave() = default;
+    constexpr view_interleave(view_interleave const & rhs) = default;
+    constexpr view_interleave(view_interleave && rhs) = default;
+    constexpr view_interleave & operator=(view_interleave const & rhs) = default;
+    constexpr view_interleave & operator=(view_interleave && rhs) = default;
+    ~view_interleave() = default;
+
+    /*!\brief Construct from a standard RandomAccessRange urange and inserted_range.
+     * \param[in] _urange The underlying range.
+     * \param[in] _step_size The step size for the insertion.
+     * \param[in] _inserted_range The range to be inserted.
+     */
+    view_interleave(urng_t _urange, size_t const _step_size, inserted_rng_t _inserted_range) :
+        urange{_urange}, step_size{_step_size}, inserted_range{_inserted_range}
+    {}
+
+    /*!\brief Construct from a ViewableRange urange and inserted_range by wrapping in a view::all.
+     * \param[in] _urange The underlying range.
+     * \param[in] _step_size The step size for the insertion.
+     * \param[in] _inserted_range The range to be inserted.
+     */
+    template <std::ranges::ViewableRange orng_t, std::ranges::ViewableRange oirng_t>
+    view_interleave(orng_t _urange, size_t const _step_size, oirng_t _inserted_range) :
+        view_interleave{view::all(_urange), _step_size, view::all(_inserted_range)}
```

Comment:
then also wrap _range in `std::forward<orng_t>(_range=`
 (same for inserted_range)